### PR TITLE
fragment_course_home.xml changed to use constraintLayout

### PR DESCRIPTION
**Pull request by kulloveth**
I tried as much as I could to reduce nested layouts but the way the views  inside the cardViews are positioned retricted that. @ujjwalagrawal17  review and give feedback. This is a fix to  #100 

Before Picture
fragmentCourse

After Picture

fragmentCourseAfter

### CODE REVIEW

File: app/src/main/res/layout/fragment_course_home.xml [hunk 1]
```diff
@@ -20,456 +20,479 @@
...
-                    android:layout_marginBottom="@dimen/padding_8dp">
 
-                    <TextView
-                        android:layout_width="match_parent"
-                        android:layout_height="wrap_content"
-                        android:text="@string/course_title"
-                        android:textColor="@color/colorAccent"
-                        android:textSize="@dimen/text_size_24sp" />
+                <TextView
+                    android:id="@+id/textView50"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID: Ex. - tv_course_title_label

File: app/src/main/res/layout/fragment_course_home.xml [hunk 2]
```diff
@@ -20,456 +20,479 @@
...
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:layout_marginTop="40dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimensions. Put it in dimens file.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 3]
```diff
@@ -20,456 +20,479 @@
...
-                        android:layout_marginStart="@dimen/padding_16dp"
-                        android:background="@drawable/button_border_accent_color"
-                        android:text="@string/btn_cancel"
-                        android:textAllCaps="false"
-                        android:textColor="@color/colorAccent"
-                        android:textSize="@dimen/text_size_16sp" />
-                </LinearLayout>
-
-            </LinearLayout>
+                    android:layout_marginStart="56dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 4]
```diff
@@ -20,456 +20,479 @@
...
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintStart_toStartOf="@+id/button3"
+                    app:layout_constraintTop_toTopOf="parent" />
 
-                </LinearLayout>
 
-                <LinearLayout
-                    android:layout_width="match_parent"
+                <Button
+                    android:id="@+id/button3"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID: Ex. - btn_join

File: app/src/main/res/layout/fragment_course_home.xml [hunk 5]
```diff
@@ -20,456 +20,479 @@
...
+                    android:text="@string/btn_join"
+                    android:textAllCaps="false"
+                    android:textColor="@color/white"
+                    android:textSize="@dimen/text_size_16sp"
+                    app:layout_constraintBottom_toBottomOf="@+id/button4"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="@+id/button4" />
+
+                <Button
+                    android:id="@+id/button4"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID: Ex. - btn_cancel

File: app/src/main/res/layout/fragment_course_home.xml [hunk 6]
```diff
@@ -20,456 +20,479 @@
...
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginStart="16dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Use padding_16dp dimens.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 7]
```diff
@@ -20,456 +20,479 @@
...
+
+        <TextView
+            android:id="@+id/tv_count_articles_created"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginTop="8dp"
+            android:layout_marginEnd="16dp"
+            android:layout_marginRight="16dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Use padding_8dp, padding_16dp, Also marginStart and marginLeft Value should be same.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 8]
```diff
@@ -20,456 +20,479 @@
...
+            android:text="@string/_45"
+            android:textColor="@color/colorAccent"
+            android:textSize="@dimen/text_size_30sp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@+id/tv_count_articles_created"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintVertical_chainStyle="packed" />
+
+        <TextView
+            android:id="@+id/textView52"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 9]
```diff
@@ -20,456 +20,479 @@
...
+            android:padding="@dimen/padding_8dp"
+            android:text="@string/_45"
+            android:textColor="@color/colorAccent"
+            android:textSize="@dimen/text_size_30sp"
+            app:layout_constraintEnd_toEndOf="@id/textView51"
+            app:layout_constraintStart_toStartOf="@id/textView51"
+            app:layout_constraintTop_toBottomOf="@+id/textView51" />
+
+        <TextView
+            android:id="@+id/textView53"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 10]
```diff
@@ -20,456 +20,479 @@
...
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginEnd="16dp"
+            android:layout_marginRight="16dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 11]
```diff
@@ -20,456 +20,479 @@
...
+            android:gravity="center"
+            android:padding="@dimen/padding_8dp"
+            android:text="Total Edits"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode strings.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 12]
```diff
@@ -20,456 +20,479 @@
...
+            android:padding="@dimen/padding_8dp"
+            android:text="@string/_45"
+            android:textColor="@color/colorAccent"
+            android:textSize="@dimen/text_size_30sp"
+            app:layout_constraintEnd_toEndOf="@id/textView52"
+            app:layout_constraintStart_toStartOf="@id/textView52"
+            app:layout_constraintTop_toBottomOf="@+id/textView52" />
+
+        <TextView
+            android:id="@+id/textView54"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 13]
```diff
@@ -20,456 +20,479 @@
...
+        <TextView
+            android:id="@+id/tv_count_words_added"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginTop="@dimen/padding_16dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginRight="8dp"
+            android:layout_marginBottom="1dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens

File: app/src/main/res/layout/fragment_course_home.xml [hunk 14]
```diff
@@ -20,456 +20,479 @@
...
+            android:padding="@dimen/padding_8dp"
+            android:text="@string/_45"
+            android:textColor="@color/colorAccent"
+            android:textSize="@dimen/text_size_30sp"
+            app:layout_constraintEnd_toEndOf="@id/textView53"
+            app:layout_constraintStart_toStartOf="@id/textView53"
+            app:layout_constraintTop_toBottomOf="@id/textView53" />
+
+        <TextView
+            android:id="@+id/textView55"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 15]
```diff
@@ -20,456 +20,479 @@
...
+            android:padding="@dimen/padding_8dp"
+            android:text="@string/_45"
+            android:textColor="@color/colorAccent"
+            android:textSize="@dimen/text_size_30sp"
+            app:layout_constraintEnd_toEndOf="@id/textView54"
+            app:layout_constraintStart_toStartOf="@id/textView54"
+            app:layout_constraintTop_toBottomOf="@id/textView54" />
+
+        <TextView
+            android:id="@+id/textView56"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 16]
```diff
@@ -20,456 +20,479 @@
...
 
-            </LinearLayout>
+        <TextView
+            android:id="@+id/tv_count_commons_uploads"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginRight="8dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 17]
```diff
@@ -20,456 +20,479 @@
...
+
+        <TextView
+            android:id="@+id/textView57"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginLeft="8dp"
+            android:layout_marginEnd="8dp"
+            android:layout_marginRight="8dp"
+            android:layout_marginBottom="8dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 18]
```diff
@@ -20,456 +20,479 @@
...
+            android:padding="@dimen/padding_8dp"
+            android:text="@string/_45"
+            android:textColor="@color/colorAccent"
+            android:textSize="@dimen/text_size_30sp"
+            app:layout_constraintEnd_toStartOf="@id/textView56"
+            app:layout_constraintStart_toEndOf="@id/textView55"
+            app:layout_constraintTop_toBottomOf="@id/textView56" />
+
+        <TextView
+            android:id="@+id/textView57"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 19]
```diff
@@ -20,456 +20,479 @@
...
+                android:textColor="@color/colorAccent"
+                android:textSize="@dimen/text_size_24sp"
+                android:textStyle="bold" />
+
+            <TextView
+                android:id="@+id/tv_course_description"
+                android:layout_width="match_parent"
+                android:layout_height="match_parent"
+                android:layout_marginStart="8dp"
+                android:layout_marginTop="48dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens.

File: app/src/main/res/layout/fragment_course_home.xml [hunk 20]
```diff
@@ -20,456 +20,479 @@
...
-                android:layout_weight="1"
-                android:orientation="vertical"
-                android:padding="@dimen/padding_8dp">
 
                 <TextView
-                    android:id="@+id/tv_count_words_added"
-                    android:layout_width="match_parent"
-                    android:layout_height="match_parent"
-                    android:gravity="center"
+                    android:id="@+id/textView37"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 21]
```diff
@@ -20,456 +20,479 @@
...
-                android:layout_weight="1"
-                android:orientation="vertical"
-                android:padding="@dimen/padding_8dp">
 
                 <TextView
-                    android:id="@+id/tv_count_article_views"
-                    android:layout_width="match_parent"
-                    android:layout_height="match_parent"
-                    android:gravity="center"
+                    android:id="@+id/textView38"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 22]
```diff
@@ -20,456 +20,479 @@
...
-                android:layout_weight="2"
-                android:orientation="vertical"
-                android:padding="@dimen/padding_8dp">
 
                 <TextView
-                    android:id="@+id/tv_count_commons_uploads"
-                    android:layout_width="match_parent"
-                    android:layout_height="match_parent"
-                    android:gravity="center"
+                    android:id="@+id/textView39"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 23]
```diff
@@ -20,456 +20,479 @@
...
-                android:layout_height="wrap_content"
-                android:background="@color/colorPrimary"
-                android:orientation="vertical"
-                android:padding="@dimen/padding_8dp">
 
                 <TextView
-                    android:id="@+id/tv_course_title"
-                    android:layout_width="match_parent"
-                    android:layout_height="match_parent"
+                    android:id="@+id/textView40"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 24]
```diff
@@ -20,456 +20,479 @@
...
-                android:layout_width="match_parent"
-                android:layout_height="wrap_content"
-                android:background="@color/colorPrimary"
-                android:orientation="vertical"
-                android:padding="@dimen/padding_8dp">
 
                 <TextView
-                    android:layout_width="match_parent"
-                    android:layout_height="match_parent"
+                    android:id="@+id/textView41"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Please use relevant ID

File: app/src/main/res/layout/fragment_course_home.xml [hunk 25]
```diff
@@ -20,456 +20,479 @@
...
-                        android:layout_width="wrap_content"
-                        android:layout_height="match_parent"
-                        android:padding="@dimen/padding_8dp"
-                        android:text="@string/description"
-                        android:textAlignment="textStart" />
-
-                </LinearLayout>
-
-            </LinearLayout>
+                    android:layout_marginBottom="8dp"
```
Comment (ujjwalagrawal17) [inline app/src/main/res/layout/fragment_course_home.xml]: Don't hardcode dimens.

**Comment by ujjwalagrawal17** [comment]
Also please have a look at https://github.com/WikiEducationFoundation/apps-android-wikiedudashboard/blob/master/PULL_REQUEST_TEMPLATE.md

**Comment by kulloveth** [comment]
@ujjwalagrawal17  kindly review changes made

**Comment by kulloveth** [comment]
@ujjwalagrawal17  i just reviewed the pull_request template. Will adhere to that onwards

**Comment by ujjwalagrawal17** [comment]
> @ujjwalagrawal17 i just reviewed the pull_request template. Will adhere to that onwards. about the travel CI build, have not used it before will read up on it. any direct link about that will be appreciated

We were using a Travis CI build earlier while reviewing PRs. I will try to activate it again this week.

**Comment by kulloveth** [comment]
@ujjwalagrawal17  have you done a final review on this?

**Comment by kulloveth** [comment]
@ujjwalagrawal17  I will check it out and fix it in my next PR

**Pull request merged**